core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -153,6 +153,49 @@ func (provider *VersionedDBProvider) Close() {
 	provider.redoLoggerProvider.close()
 }
 
+// Drop drops the couch dbs and redologger data for the channel.
+// It is not an error if a database does not exist.
+func (provider *VersionedDBProvider) Drop(channelName string) error {
+	metadataDBName := constructMetadataDBName(channelName)
+	couchDBDatabase := couchDatabase{couchInstance: provider.couchInstance, dbName: metadataDBName, indexWarmCounter: 1}
+	_, couchDBReturn, err := couchDBDatabase.getDatabaseInfo()
+	if couchDBReturn != nil && couchDBReturn.StatusCode == 404 {
+		// db does not exist
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	metadataDB, err := createCouchDatabase(provider.couchInstance, metadataDBName)
```

Comment:
We should rename `createCouchDatabase` to `getOrCreateCouchDatabase` but perhaps in a separate PR.